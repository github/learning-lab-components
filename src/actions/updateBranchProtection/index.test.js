const updateBranchProtection = require('./')
const mockContext = require('../../tests/mockContext')

describe('updateBranchProtection', () => {
  let context

  beforeEach(() => {
    context = mockContext({}, {
      repos: {
        updateBranchProtection: jest.fn()
      }
    })
  })

  it('updates branch protection on master by default', async () => {
    await updateBranchProtection(context, {})
    expect(context.github.repos.updateBranchProtection).toHaveBeenCalled()
    expect(context.github.repos.updateBranchProtection.mock.calls).toMatchSnapshot()
  })

  it('updates branch protection on the provided branch', async () => {
    await updateBranchProtection(context, { branch: 'a-branch' })
    expect(context.github.repos.updateBranchProtection).toHaveBeenCalled()
    expect(context.github.repos.updateBranchProtection.mock.calls).toMatchSnapshot()
  })
})
