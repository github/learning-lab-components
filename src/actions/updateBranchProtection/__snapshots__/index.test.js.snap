// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateBranchProtection updates branch protection on master by default 1`] = `
Array [
  Array [
    Object {
      "branch": "master",
      "enforce_admins": true,
      "owner": "JasonEtco",
      "repo": "example",
      "required_pull_request_reviews": Object {},
      "required_status_checks": null,
      "restrictions": null,
    },
  ],
]
`;

exports[`updateBranchProtection updates branch protection on the provided branch 1`] = `
Array [
  Array [
    Object {
      "branch": "a-branch",
      "enforce_admins": true,
      "owner": "JasonEtco",
      "repo": "example",
      "required_pull_request_reviews": Object {},
      "required_status_checks": null,
      "restrictions": null,
    },
  ],
]
`;
